srcCpp/Infrastructure_micro.h

```diff
@@ -159,16 +159,59 @@ class NDDSUtility {
     #endif
     };
 
+    /* Get the time cost to call getTimeUsec a number of times equal to iteration */
+    static unsigned long long get_time_overhead(unsigned int iterations) {
+
+        PerftestClock clock = PerftestClock::getInstance();
+        unsigned long long init = 0, fin = 0, overhead = 0;
+
+        init = clock.getTimeUsec();
+        for (int i = 0; i < iterations; i++) {
+            clock.getTimeUsec();
+        }
+        fin = clock.getTimeUsec();
+
+        return fin - init;
+    }
+
     /*e \dref_Utility_spin */
     static void spin(DDS_UnsignedLongLong spinCount){
-        /* spin option is not implemented */
-        printf("[ERROR]: Spin is not implemented for Micro");
+        NDDS_Utility_spin(spinCount);
     };
 
-    static DDS_UnsignedLongLong get_spin_per_microsecond() {
-        printf("[ERROR]: get_spin_per_microsecond is not implemented for Micro");
-        return -1;
-    };
+
+    static DDS_UnsignedLongLong
+    get_spin_per_microsecond(unsigned int precision = 100)
+    {
+        /* Same default values used by DDS */
+        unsigned int spinCount = 20000;
+
+        unsigned long long init = 0, fin = 0, usec = 0;
+        unsigned long long iterations = 0;
+        unsigned long long overhead = 0;
+
+        PerftestClock clock = PerftestClock::getInstance();
+
+        while((int)(usec - overhead) < (int)precision){
+
+            init = clock.getTimeUsec();
+
+            NDDS_Utility_spin(spinCount); //Same numbers of spin iters as DDS
+
+            fin = clock.getTimeUsec();
+
+            usec += fin - init;
+
+            iterations++;
+            overhead = get_time_overhead(iterations);
+        }
+
+        /* Measure the clock.getTimeUsec() overhead */
+        usec -= overhead;
+
+        return (float)(iterations * spinCount) / usec;
```

Comment:
Add a ward for usec, in certain OS it can be "0", due to unreliable clocks with low resolution